scripts/templates/params.hpp.mako

```diff
@@ -277,6 +298,24 @@ inline std::ostream &operator<<(std::ostream &os, const ${obj['type']} ${th.make
     os << "}";
     return os;
 }
+%elif re.match(r"union", obj['type']) and obj['name']:
+<% tag = findUnionTag(obj) %>
+inline void ${x}_params::serializeUnion(
+    std::ostream &os,
+    const ${obj['type']} ${th.make_type_name(n, tags, obj)} params,
+    const ${tag['type']} ${th.make_type_name(n, tags, tag)} tag
+){
+    switch(tag){
+%for mem in obj['members']:
+    case ${th.subt(n, tags, mem['tag'])}:
+        os << params.${mem['name']};
```

Comment:
This should use `line` function to print members so that all union member types are handled correctly:
```
    <%
        params_dict = dict()
        for item in obj['params']:
            iname = th._get_param_name(n, tags, item)
            itype = th._get_type_name(n, tags, obj, item)
            params_dict[iname] = itype
    %>
        switch (tag) {
              %for item in obj['members']:
                   case ${th.subt(n, tags, item['tag'])}:
                      ${line(item, 0, False, params_dict)}
                      break;
               %endfor
                   default:
                      os << "<unknown>";
                      break;
       }
```

Reply:
Fixed 👍